src/server/system_services/account_server.js

```diff
@@ -712,6 +713,79 @@ async function add_external_connection(req) {
     });
 }
 
+async function edit_external_connection_credentials(req) {
+    const { name, identity, secret } = req.rpc_params;
+    const { account } = req;
+    const connection_to_edit = cloud_utils.find_cloud_connection(account, name);
+    const { endpoint_type, cp_code, auth_method, endpoint } = connection_to_edit;
+    const new_connection = _.cloneDeep(connection_to_edit);
+    const check_connection = {
+        name,
+        identity,
+        secret,
+        endpoint_type,
+        cp_code,
+        auth_method,
+        endpoint,
+    };
+    try {
+        const reply = await _check_external_connection(check_connection);
+        // TODO: Maybe handle differently
+        if (reply.status !== 'SUCCESS') {
+            throw new RpcError('CREDENTIALS_INVALID', `Credentials are not valid ${name}`);
+        }
+    } catch (error) {
+        // TODO: Maybe handle differently
+        dbg.error('_check_external_connection had error', error);
+        throw new RpcError('CREDENTIALS_INVALID', `Credentials are not valid ${name}`);
+    }
+    new_connection.access_key = identity;
+    new_connection.secret_key = secret;
+
+    const account_updates = {
+        _id: req.account._id,
+        sync_credentials_cache: _.filter(account.sync_credentials_cache,
+            connection => (connection.name !== name))
+    };
+
+    account_updates.sync_credentials_cache.push(new_connection);
```

Comment:
what about the old connection? aren't there two connections with the same name now?

Reply:
Notice above when I declare the account_updates, I remove the old connection using the filter operation.
Which means that I only leave the connections without the name that we want to edit.